### DISCUSSION TITLE
Temperature bias correction

### DISCUSSION THREAD

**Discussion by ArcticSnow** [ArcticSnow/TopoPyScale#65] [Ideas]
Hey @joelfiddes @yelizy @krisaalstad ,
Do you know what would be the best strategy to bias correct the downscaled temperatures when using `toposub` given there is handful of weather stations in the domain?

Some ideas:
1. Applying an overall mean bias to all clusters
2. fit a plane or compute a 2D interpolation / kriggin field in between the point bias can be estimated 
3. A quantile mapping approach (https://journals.ametsoc.org/view/journals/clim/26/6/jcli-d-12-00821.1.xml) or available in [xclim](https://xclim.readthedocs.io/en/stable/notebooks/sdba-advanced.html)

**Comment by krisaalstad** [comment]
Hey Simon, I think there is another complementary field related to this known as empirical-statistical downscaling. The guys at metno, especially Rasmus Benestad (this review-like paper somehow got reject but it looks interesting despite the strange name: 
 https://gmd.copernicus.org/preprints/gmd-2021-176/ , it's probably also a good source of references), work a lot on this. Although it breaks somewhat with the original philosophy of TopoSCALE that it should be applicable anywhere on Earth (even without ground observations), it would of course be silly not to use ground observations at least for a bias correction if you do have them. We should probably think about going towards a hybrid topography-based + ESD tool at some point. 

I know some people at NERSC in Bergen (Richard Davy, see his KrigR tool) that use kriging methods for downscaling, and I think they would be interested in developing such hybrid tools. You could consider reaching out to them. Still, with sparse observations I would be quite curious about what is validating what when comparing a kriged field to TopoSCALE. I think TopoSCALE would be better at capturing local variability that kriging can never really resolve (since it is limited by the density of the observing network). Still, kriging could be used to do a larger scale bias correction which would probably make a lot of sense, while keeping the spatial patterns from TopoSCALE. I think you have outlined most of the methods quite nicely, but you are maybe missing (multiple) linear regression which is usually the first thing people try for a simple bias correction that allows for (somewhat risky) extrapolation of the bias to unobserved areas. Note that kriging is actually a form of linear regression too, but it allows you to include more covariates and is based on more explicit modeling of spatial variability. Happy to discuss more.

**Reply by joelfiddes** [reply]
just a small comment from my side: I think this is really an important use case as in data-poor regions there are often a handful of stations and I often get this request - although even without using them for downscaling they can represent critical evaluation data (which is not possible if used).

Also we need to be super critical about what those point measurements represent in terms of station siting, bias in station distribution (represent valley locations where era5 is more biased?) and also the measurements themselves - unheated etc. I know you alll know these issues well but just thought get it down in the discussion

**Reply by ArcticSnow** [reply]
Thank you guys for your inputs.

What Joel describes is my exact case. So my strategy is 
1. downscale to the weather station points and estimate bias via linear regression for each of them, 
2. spatialize the bias estimates, 
3. downscale spatially with topo_sub, and 
4. apply the spatial bias correction to the spatially downscaled timeseries (assuming this is stationary in time). 

I'll keep you updated on my progress.
